pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/AvroSchemaCache.java

```diff
@@ -65,7 +66,7 @@ public AvroSchemaCache(SchemaRegistryClient schemaRegistryClient) {
             org.apache.avro.Schema schema = schemaRegistryClient.getById(schemaId);
             String definition = schema.toString(false);
             log.info("Schema {} definition {}", schemaId, definition);
-            SchemaInfo schemaInfo = SchemaInfo.builder()
+            SchemaInfo schemaInfo = SchemaInfoImpl.builder()
```

Comment:
This is the code of a Source, we cannot use Impl classes.
They are not available here directly

Reply:
I think this was already pulling in pulsar-common

Reply:
yes, but this is not good anyway, because if you develop a Source you cannot use this class.
is this Source adding pulsar-common in the dependencies ?

Reply:
it does not matter that this Source is inside the main Pulsar repo

Reply:
> is this Source adding pulsar-common in the dependencies ?

Yes, that is the case

Reply:
I think the current state is OK for now. All functionality is working with current approach. 

For a more general approach I think we should add a mechanism to detect the classloader information and identify we are in "pulsar function mode". Therefore, in `DefaultImplementation` class, we should be able to directly use the appropriate classloader, irrespective of where the access is coming from. 

I believe this is something we should be targeting for 2.9.